README.adoc

```diff
@@ -88,85 +88,122 @@ The `withGradle` wrapper should be used instead, since it also deals well with p
+
+==== Gradle environment variables
+
+===== `JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_PLUGIN_VERSION`
+
+Enables auto-injection for Gradle builds and defines which version of the https://docs.gradle.com/enterprise/gradle-plugin/[Gradle Enterprise Gradle plugin] to use.
+
+===== `JENKINSGRADLEPLUGIN_CCUD_PLUGIN_VERSION`
+
+Defines which version of the https://github.com/gradle/common-custom-user-data-gradle-plugin[Common Custom User Data Gradle plugin] to use.
+
+===== `JENKINSGRADLEPLUGIN_GRADLE_PLUGIN_REPOSITORY_URL`
+
+The URL of the repository to use to resolve the Gradle Enterprise Gradle plugin and the Common Custom User Data Gradle plugin.
+This is required if the Jenkins agents are not able to access the https://plugins.gradle.org[Gradle Plugin Portal].
+
+===== `JENKINSGRADLEPLUGIN_GRADLE_INJECTION_ENABLED_NODES`
+
+Comma separated list of node labels on which the Gradle Enterprise Gradle plugin injection should be enabled.
+By default, all nodes are enabled.
+
+===== `JENKINSGRADLEPLUGIN_GRADLE_INJECTION_DISABLED_NODES`
+
+Comma separated list of node labels on which the Gradle Enterprise Gradle plugin injection should be disabled.
+By default, all nodes are enabled.
+
+==== Maven environment variables
+
+===== `JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_EXTENSION_VERSION`
+Enables auto-injection for Maven builds and defines which version of the https://docs.gradle.com/enterprise/maven-extension[Gradle Enterprise Maven extension] to use. Currently, the value of these environment variable is ignored and is just used as a trigger to enable the auto-injection.
+
+===== `JENKINSGRADLEPLUGIN_CCUD_EXTENSION_VERSION`
+Defines which version of the https://github.com/gradle/common-custom-user-data-maven-extension[Common Custom User Data Maven Extension] to use. Currently, the value of this environment variable is ignored and is just used as a trigger to enable the auto-injection.
+
+===== `JENKINSGRADLEPLUGIN_MAVEN_INJECTION_ENABLED_NODES`
+Comma separated list of node labels where the Gradle Enterprise Maven extension injection should be enabled. By default, all nodes are enabled.
 
-In addition to the variables above you might want to set the `GRADLE_ENTERPRISE_ACCESS_KEY` variable if you're Gradle Enterprise server requires authentication for publishing Build Scans.Refer to the https://docs.gradle.com/enterprise/gradle-plugin/#via_environment_variable[Gradle Enterprise Gradle plugin manual] and the https://docs.gradle.com/enterprise/maven-extension/#via_environment_variable[Gradle Enterprise Maven Extension manual] for more information.
+===== `JENKINSGRADLEPLUGIN_MAVEN_INJECTION_DISABLED_NODES`
+Comma separated list of node labels where the Gradle Enterprise Maven extension injection should be disabled. By default, all nodes are enabled.
```

Comment:
same as above, which one wins of you set a node label in both enabled/disabled places ?

Reply:
`Disabled` check take precedence. If the node is disabled, the `Enabled` check is not performed.

Reply:
Then we should explain it